### PR TITLE
Fix revapi release check

### PR DESCRIPTION
**Pull request by yunikkk**
### Summary of changes
Fixed revapi check both for release and main branches. Currently it 
- fails on PRs to release branch picking incorrect `android-v10.6.0` release - [the build](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/6793/workflows/aad95f19-6525-40e9-b090-44692f009e5f/jobs/57080)
```
$1:
$2:./plugin-compass/build/outputs/aar/plugin-compass-release.aar
$3:compass
$4:android-v10.6.0
Path to previous version of aar is not set, using 10.6.0
aar path is: s3://mapbox-api-downloads-production/v2/mobile-maps-android-compass/releases/android/10.6.0/maven/maps-compass-10.6.0.aar, checking file
```

- succeedes without any actual comparison on PRs to the main branch - [the build](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/6781/workflows/50301da3-815e-4f4a-9fa1-c33a39d85529/jobs/56967)
```
$1:
$2:./plugin-viewport/build/outputs/aar/plugin-viewport-release.aar
$3:viewport
$4:v10.8.0
Path to previous version of aar is not set, using 
aar path is: s3://mapbox-api-downloads-production/v2/mobile-maps-android-viewport/releases/android//maven/maps-viewport-.aar, checking file
s3://mapbox-api-downloads-production/v2/mobile-maps-android-viewport/releases/android//maven/maps-viewport-.aar doesn't exist on s3, skipping the check
```

After the change script correctly compares the previous release, e.g. [build](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/6796/workflows/041ddb7b-4828-4e14-b947-c5d1c23b3f56/jobs/57187) with `10.9.0-rc.1`
```
$1:
$2:./plugin-viewport/build/outputs/aar/plugin-viewport-release.aar
$3:viewport
$4:v10.9.0-rc.1
Path to previous version of aar is not set, using 10.9.0-rc.1
aar path is: s3://mapbox-api-downloads-production/v2/mobile-maps-android-viewport/releases/android/10.9.0-rc.1/maven/maps-viewport-10.9.0-rc.1.aar, checking file
Downloading file from s3

```

Note that job still fails with bunch of breaking changes within the PR, but all seem to be related to internal apis, so can be ignored

```
======================== animation ========================
Analysis results
----------------

Old API: tmp.OeBsOlxGhh
New API: tmp.6Cq5npkpLL
old: method boolean com.mapbox.maps.plugin.animation.HighLevelAnimatorSet::getStarted()
new: <none>
java.method.removed: Method was removed.
BINARY: BREAKING, SOURCE: BREAKING

old: method void com.mapbox.maps.plugin.animation.HighLevelAnimatorSet::setStarted(boolean)
new: <none>
java.method.removed: Method was removed.
BINARY: BREAKING, SOURCE: BREAKING

======================== gestures ========================
Analysis results
----------------

Old API: tmp.XrSV6hKpQp
New API: tmp.cgX7mHNChE
old: method void com.mapbox.maps.plugin.gestures.GesturesPluginImpl::<init>(android.content.Context, android.util.AttributeSet, float, android.os.Handler)
new: method void com.mapbox.maps.plugin.gestures.GesturesPluginImpl::<init>(android.content.Context, android.util.AttributeSet, float, android.os.Handler, android.os.Handler)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
BINARY: BREAKING, SOURCE: BREAKING

old: method boolean com.mapbox.maps.plugin.gestures.GesturesPluginImpl::handleFlingEvent$plugin_gestures_release(android.view.MotionEvent, android.view.MotionEvent, float, float)
new: method boolean com.mapbox.maps.plugin.gestures.GesturesPluginImpl::handleFlingEvent$plugin_gestures_release(android.view.MotionEvent, float, float)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
BINARY: BREAKING, SOURCE: BREAKING

======================== sdk ========================
Analysis results
----------------

Old API: tmp.fPllR71zuI
New API: tmp.1v4qILN4bK
old: field com.mapbox.maps.renderer.MapboxWidgetRenderer.TAG
new: field com.mapbox.maps.renderer.MapboxWidgetRenderer.TAG
java.field.visibilityReduced: Visibility was reduced from 'public' to 'private'.
BINARY: BREAKING, SOURCE: BREAKING

======================== sdk-base ========================
Analysis results
----------------

Old API: tmp.9ZZe5NSaLK
New API: tmp.o1AVnbDrek
old: method com.mapbox.maps.plugin.LocationPuck2D com.mapbox.maps.plugin.LocationPuck2D::copy(android.graphics.drawable.Drawable, android.graphics.drawable.Drawable, android.graphics.drawable.Drawable, java.lang.String)
new: method com.mapbox.maps.plugin.LocationPuck2D com.mapbox.maps.plugin.LocationPuck2D::copy(android.graphics.drawable.Drawable, android.graphics.drawable.Drawable, android.graphics.drawable.Drawable, java.lang.String, float)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
BINARY: BREAKING, SOURCE: BREAKING
```

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: scripts/java-api-check.sh [hunk 1]
```diff
@@ -63,14 +68,14 @@ if [[ -z $RELEASE_TAG ]]; then
   fi
   echo "aar path is: $AAR_PATH, checking file"
   if [[ -z $(aws s3 ls "$AAR_PATH") ]]; then
-    echo "$AAR_PATH doesn't exist on s3, skipping the check"
-    exit 0
+    echo "$AAR_PATH doesn't exist on s3!"
+    exit 1
```
Comment (yunikkk) [inline scripts/java-api-check.sh:72]: Fail if previous binary is not found on S3. Will break the check when the artifact is first released, but guarantee we're not silently ignoring missing versions as it happened for a long time.

File: scripts/java-api-check.sh [hunk 2]
```diff
@@ -50,27 +55,30 @@ PREVIOUS_RELEASE_DIR=$(dirname "${PREVIOUS_RELEASE}")
...
 LAST_STABLE_VERSION_TAG=$4 #v10.3.0
 LAST_STABLE_VERSION_TAG_ARRAY=($LAST_STABLE_VERSION_TAG)
-LAST_STABLE_VERSION=${LAST_STABLE_VERSION_TAG_ARRAY[0]:9}
+LAST_STABLE_VERSION=${LAST_STABLE_VERSION_TAG_ARRAY[0]:1}
 
 RELEASE_TAG=${5-""}
 if [[ -z $RELEASE_TAG ]]; then
-  echo "Path to previous version of aar is not set, using ${LAST_STABLE_VERSION}"
+  echo "Downloading ${LAST_STABLE_VERSION} release from the SDK Registry to check api compatibility with revapi."
   AAR_PATH="s3://mapbox-api-downloads-production/v2/mobile-maps-android-"${MODULE_NAME}"/releases/android/"${LAST_STABLE_VERSION}"/maven/maps-"${MODULE_NAME}"-"${LAST_STABLE_VERSION}".aar"
```
Comment (pengdev) [inline scripts/java-api-check.sh:63]: if possible I'd prefer not interacting with S3, and use our production endpoint instead, as this is a public repo.

**Review by pengdev — APPROVED** [review]
LGTM % nit

**Review by ank27 — APPROVED** [review]

File: scripts/java-api-check.sh (hunk 2 above)
Comment (yunikkk) [reply scripts/java-api-check.sh:63]: Fair, however I'd do it in a separate PR/issue since it seems out of scope for thix fix

Comment (yunikkk) [reply scripts/java-api-check.sh]: Replaced with https endpoint

**Pull request merged**